nimterop/cimport.nim

```diff
@@ -170,6 +173,28 @@ macro cSkipSymbol*(skips: varargs[string]): untyped =
   for skip in skips:
     gStateCT.symOverride.add skip.strVal
 
+macro cPlugin*(body): untyped =
+  ## When `cOverride() <cimport.html#cOverride.m,>`_ and `cSkipSymbol() <cimport.html#cSkipSymbol.m%2Cvarargs[string]>`_
+  ## are not adequate, the `cPlugin() <cimport.html#cPlugin.m,>`_ macro can be used to customize the generated Nim output.
+  ## The following callbacks are available at this time.
+  ##
+  ## .. code-block:: nim
+  ##
+  ##    cPlugin:
+  ##      import strutils
+  ##
+  ##      proc onSymbol*(sym: string): string {.exportc, dynlib.} =
+  ##        return sym.strip(chars={'_'})
+
+  let
+    data = body.repr
+    path = getTempDir() / "nimterop" & ($data.hash() & ".nim")
```

Comment:
could we use `nimcache` instead of `getTempDir()` ? a few reasons; one of them is `/tmp` is wiped upon reboot and maybe we dont' wanna invalidate our cache just because of machine reboot; i think there are other reasons
but need to check the right API to get the right nimcache

Reply:
Doesn't look like nimcache is accessible other than through nimscript.

The code is in `compiler\options.nim` but not exposed to programs.

Reply:
[RESOLVED]